build-logic/jvm/src/main/kotlin/gradlebuild.unittest-and-compile.gradle.kts

```diff
@@ -229,10 +229,16 @@ fun configureTests() {
         }
 
         useJUnitPlatform()
+
+        distribution.enabled.set(true)
```

Comment:
Does this work for our performance tests as well? Can we run unit tests always with our scheduler, so we dogfood this as well?

Reply:
I've fixed the majority (maybe all) of the failures that were caused by mishandling (on the part of the test filtering plugin) of multiple test engines running the same tests. 

I'm afraid using the Test Distribution plugin in local-only mode might still skew performance tests. I'm going to run some on the branch and see. What is your intuition here @wolfs?

Reply:
I would keep test distribution disabled for performance tests right now and try it out there in a separate PR.

Reply:
And let's keep test selection/test distribution disabled for unit test tasks.

Reply:
Ok, I've made these changes.